service/src/main/kotlin/fi/espoo/evaka/invoicing/service/VoucherValueDecisionGenerator.kt

```diff
@@ -179,13 +177,18 @@ private fun generateNewValueDecisions(
 
             voucherChildPlacement.let { placement ->
                 val familyIncomes = family.partner?.let { listOf(income, partnerIncome) } ?: listOf(income)
-                val baseCoPayment = calculateBaseFee(price, family.getSize(), familyIncomes + listOfNotNull(childIncome))
-                val baseValue = getBaseValue(period, voucherChild.dateOfBirth, voucherValue)
-                val value =
-                    if (placement == null || serviceVoucherUnits.none { unit -> unit == placement.unitId }) 0
-                    else calculateVoucherValue(baseValue, capacityFactor, placement.serviceNeed.voucherValueCoefficient)
+                val baseCoPayment =
+                    calculateBaseFee(price, family.getSize(), familyIncomes + listOfNotNull(childIncome))
+                val voucherValue =
+                    if (placement == null || serviceVoucherUnits.none { unit -> unit == placement.unitId }) null
+                    else {
+                        val voucherValue =
+                            voucherValues[placement.serviceNeed.optionId]?.find { it.validity.contains(period) }
+                                ?: error("Missing voucher value for period ${period.start} - ${period.end}, cannot generate voucher value decision")
```

Comment:
Service need option id should be logged here. Otherwise it's hard to know which voucher value is missing.

Reply:
Yes, I agree.

Reply:
Fixed